### ISSUE TITLE
STB Reason: can't fopen/STB Reason: unknown image type Cannot load image - tried other solution

### ISSUE THREAD

**Issue by sbanerj2** [pjreddie/darknet#1504]
I am getting the following error for some images (custom dataset) as a result of which the training stops.
(I am using /path/to/ as an abbreviation of my absolute path. All the files and images exist.)

```
Loading weights from ./darknet53.conv.74...Done!
Cannot load image "/path/to/Object_Detection/Ground/JPEGImages/Training/6/um_50_na_sunny_sony_0_384.jpg"
STB Reason: can't fopen
Cannot load image "/path/to/Object_Detection/Ground/JPEGImages/Training/6/um_50_na_sunny_sony_0_41.jpg"
STB Reason: unknown image type
Cannot load image "/path/to/Object_Detection/Ground/JPEGImages/Training/3/golf_40_na_cloudy_sony_0_273.jpg"
STB Reason: unknown image type

```
My images are present in the location too and their corresponding text files are in the following location:  
/path/to/Object_Detection/Ground/labels/Training/...

The format of the folders are like Pascal VOC

here's a part of the makefile:
GPU=1
CUDNN=0
OPENCV=1
OPENMP=0
DEBUG=0

And here's my ground.data:
```
classes= 19
train  = /path/to/Object_Detection/ground_train.txt
valid  = /path/to/Object_Detection/ground_val.txt
names = /path/to/darknet/data/ground.names
backup = /path/to/darknet/backup/ground/

```
Here's the command I am using:

``
./darknet detector train /path/to/darknet/cfg/ground.data /path/to/darknet/cfg/yolov3-ground.cfg ./darknet53.conv.74 > /path/to/darknet/scripts/ground-train.log
``

Here's how my  ground_train.txt (has all absolute path for images. only a portion of the entire file is shown) looks :
```
/path/to/Object_Detection/Ground/JPEGImages/Training/0/analog_50_180_sunny_sony_0_0.jpg
/path/to/Object_Detection/Ground/JPEGImages/Training/0/analog_50_180_sunny_sony_0_1.jpg
/path/to/Object_Detection/Ground/JPEGImages/Training/0/analog_50_180_sunny_sony_0_2.jpg
...
```

All the files exists and are valid. I am not sure what's wrong. It works for other files in the training dataset but only stops for these. I checked whether the images are of correct format, and they seem ok. 

Here's the output after 14 epochs, my training stops after this:
```
14: 573.572327, 927.539124 avg, 0.000000 rate, 1636.650012 seconds, 896 images
Loaded: 0.000064 seconds
: 0.219096, Class: 0.386432, Obj: 0.530981, No Obj: 0.511523, .5R: 0.000000, .75R: 0.000000,  count: 2
Region 106 Avg IOU: 0.011220, Class: 0.282754, Obj: 0.311355, No Obj: 0.468680, .5R: 0.000000, .75R: 0.000000,  count: 1
Region 82 Avg IOU: 0.279148, Class: 0.148486, Obj: 0.629275, No Obj: 0.458073, .5R: 0.000000, .75R: 0.000000,  count: 1
Region 94 Avg IOU: 0.075961, Class: 0.271695, Obj: 0.082233, No Obj: 0.514065, .5R: 0.000000, .75R: 0.000000,  count: 2
Region 106 Avg IOU: 0.352088, Class: 0.543684, Obj: 0.793743, No Obj: 0.471246, .5R: 0.000000, .75R: 0.000000,  count: 1
Region 82 Avg IOU: 0.165622, Class: 0.815933, Obj: 0.489756, No Obj: 0.457557, .5R: 0.000000, .75R: 0.000000,  count: 1
Region 94 Avg IOU: 0.053709, Class: 0.664887, Obj: 0.541732, No Obj: 0.512764, .5R: 0.000000, .75R: 0.000000,  count: 2
Region 106 Avg IOU: 0.073817, Class: 0.247052, Obj: 0.413561, No Obj: 0.469873, .5R: 0.000000, .75R: 0.000000,  count: 1
14: 573.572327, 927.539124 avg, 0.000000 rate, 1636.650012 seconds, 896 images
Loaded: 0.000064 seconds
```

**Comment by Mps24-7uk** [comment]
I am getting the same issue Cannot load image "./darknet/dataset/z1.jpg" STB Reason: can't fopen  @sbanerj2

**Comment by SalahAdDin** [comment]
I'm also getting the same problem.

**Comment by PhilipMachineLearning** [comment]
I'm getting the same issue.

**Comment by tijnonlijn** [comment]
The same happens here on ubuntu 18.04. Now here's the funny part: after retrying 5-10 times while randomly changing the file name it suddenly works. Even when I use files that previously gave an error. And once it works it remains working without any error.  

Now I'm not really a linux expert, but could it be that there is some sort of background process that takes some time to start?

**Comment by sbanerj2** [comment]
For me the problem was compilation. I was compiling yolo with a wrong version of CUDA. Use the fork by alexeyab and make sure you have all the requirements.

**Comment by Shruthi-Sampathkumar** [comment]
I am getting the same error. Did anybody solve it?

**Comment by SalahAdDin** [comment]
@Shruthi-Sampathkumar Use the fork, it worked for me.

**Comment by AyazSaiyed** [comment]
![Screen Shot 2019-11-25 at 3 32 26 PM](https://user-images.githubusercontent.com/49748587/69530879-df0f7800-0f98-11ea-87f0-692b78a844ac.png)


**'/Users/yudiz/Downloads/darknet-master/build/darknet/x64/data/obj/000001.jpg'** 
It worked for me , issue was a extra '**space**' between two new lines !!!

**Comment by amrkulk** [comment]
Use the strip() function in python to get rid of the trailing & leading new lines. The image paths in txt files contain extra new lines.

**Comment by pushgct** [comment]
> The same happens here on ubuntu 18.04. Now here's the funny part: after retrying 5-10 times while randomly changing the file name it suddenly works. Even when I use files that previously gave an error. And once it works it remains working without any error.
> 
> Now I'm not really a linux expert, but could it be that there is some sort of background process that takes some time to start?

hi, which file did you rename to please? I too am stuck with same issue on Mac terminal.

**Comment by bzyy** [comment]
hi, you can use the python's **imghdr** to check the image Is it really jpg format

1. example
```
import imghdr
print(imghdr.what("/path/to/Object_Detection/Ground/JPEGImages/Training/6/um_50_na_sunny_sony_0_384.jpg"))
# maybe print  **webp* ! not jpg ~
```
2. now, write a sample filter script like this ~
```
import imghdr

def main():
    file_list = []
    with open("/path/to/Object_Detection/ground_train.txt") as f:
        for name in f.readlines():
            name = name.replace('\n','')
            try:
                img_type = name.split('.')[-1]
                if imghdr.what(name) in ['jpeg', img_type]:
                    print(name)
                    file_list.append(name)
            except FileNotFoundError:
                # print(name)
                continue
    with open("/path/to/Object_Detection/ground_train.txt", "w") as f:
        for name in file_list:
            f.write(name + '\n')
    
if __name__ == "__main__":
    main()
```
